numba/tests/test_np_functions.py

```diff
@@ -2503,6 +2507,117 @@ def check_pass_through(jitted, expect_same, params):
                 else:
                     check_pass_through(cfunc, True, params)
 
+    def test_repeat(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a, repeats):
+            self.assertPreciseEqual(pyfunc(a, repeats), cfunc(a, repeats))
+
+        # test array argumens
+        target_numpy_values = [
+            np.ones(1),
+            np.arange(1000),
+            np.array([[0, 1], [2, 3]]),
+            np.array([]),
+            np.array([[], []]),
+        ]
+
+        target_numpy_types = [
+            np.uint32,
+            np.int32,
+            np.uint64,
+            np.int64,
+            np.float32,
+            np.float64,
+            np.complex64,
+            np.complex128,
+        ]
+
+        target_numpy_inputs = (np.array(a,dtype=t) for a,t in
+                               itertools.product(target_numpy_values,
+                                                 target_numpy_types))
+
+        target_non_numpy_inputs = [
+            1,
+            1.0,
+            True,
+            1j,
+            [0, 1, 2],
+            (0, 1, 2),
+        ]
+        for i in itertools.chain(target_numpy_inputs, target_non_numpy_inputs):
+            check(i, repeats=0)
+            check(i, repeats=1)
+            check(i, repeats=2)
+            check(i, repeats=3)
+            check(i, repeats=100)
+
+        # check broadcasting when repeats is an array
+        one = np.arange(1)
+        check(one, repeats=np.array([0]))
+        check(one, repeats=np.array([1]))
+        check(one, repeats=np.array([2]))
+
+        two = np.arange(2)
+        check(two, repeats=np.array([0, 0]))
+        check(two, repeats=np.array([0, 1]))
+        check(two, repeats=np.array([1, 0]))
+        check(two, repeats=np.array([1, 1]))
+
+        check(two, repeats=np.array([1, 2]))
+        check(two, repeats=np.array([2, 1]))
+        check(two, repeats=np.array([2, 2]))
+        check(two, repeats=np.array([2, 2], dtype=np.int32))
+
+        check(np.arange(10), repeats=np.arange(10))
+
+    def test_repeat_exception(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+        self.disable_leak_check()
+
+        # heterogeneous list
+        with self.assertRaises(TypeError) as e:
```

Comment:
It's fine to not test heterogeneous list and nested list as input because they are being caught by boxing as invalid types opposed to being caught in the implementation presented in the PR.

Reply:
check

Reply:
awesome, that will solve a lot of CI hassle